src/plone/rest/traverse.py

```diff
@@ -18,7 +18,10 @@ def publishTraverse(self, request, name):
             obj = super(RESTTraverse, self).publishTraverse(request, name)
             if (not IContentish.providedBy(obj)
                     and not IService.providedBy(obj)):
-                raise KeyError
+                if name in ('virtual_hosting',):
```

Comment:
I would prefer testing against `isinstance(obj, VirtualHostMonster)` instead of the id, as the VHM could have an arbitrary id in theory.